fml/platform/darwin/string_range_sanitization.mm

```diff
@@ -0,0 +1,29 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/fml/platform/darwin/string_range_sanitization.h"
+
+namespace fml {
+
+NSRange RangeForCharacterAtIndex(NSString* text, NSUInteger index) {
+  if (text == nil || index > text.length) {
```

Comment:
I'd be tempted to use `index >= text.length` for consistency with Apple's APIs. It also feels a bit more intuitive: the underlying string has no character at `index == length`. 

That said, text positions _do_ allow for end-of-string positions. If every call to this is expected to have to defend against end-of-string positions, then it probably makes sense as-is, but I'd definitely document the range of valid indices in the header.

Reply:
I'll go with >=, good catch.